streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoinProcessor.java

```diff
@@ -150,7 +150,7 @@ private boolean maybeDropRecord(final Record<StreamKey, StreamValue> record) {
         // furthermore, on left/outer joins 'null' in ValueJoiner#apply() indicates a missing record --
         // thus, to be consistent and to avoid ambiguous null semantics, null values are ignored
         final TableKey mappedKey = keyMapper.apply(record.key(), record.value());
-        if (leftJoin && record.key() == null && record.value() != null) {
+        if (leftJoin && mappedKey == null && record.value() != null) {
```

Comment:
This is the actual fix.

Reply:
Could this problem also affect other join types? Have you checked?

Reply:
It's specific to stream-globalTable join.